src/azure-cli-tests/azure/cli/tests/base.py

```diff
@@ -0,0 +1,167 @@
+            name = create_random_name(prefix, length)
+            self.name_replacer.register_name_pair(name, moniker)
+            return name
+        else:
+            return moniker
+
+    def cmd(self, command, checks=None):
+        result = execute(command)
+
+        if not checks:
+            checks = []
+        elif not isinstance(checks, list):
+            checks = [checks]
+
+        if not self.skip_assert:
+            for c in checks:
+                c(result)
+
+        return result
+
+    def _process_request_recording(self, request):
+        for processor in self.recording_processors:
+            request = processor.process_request(request)
+            if not request:
+                break
+
+        return request
+
+    def _process_response_recording(self, response):
+        for processor in self.recording_processors:
+            response = processor.process_response(response)
+            if not response:
+                break
+
+        return response
+
+
+class ExecutionResult(object):
+    def __init__(self, command, expect_failure=False, in_process=True):
+        if in_process:
+            self._in_process_execute(command)
+        else:
+            self._out_of_process_execute(command)
+
+        if expect_failure and self.exit_code == 0:
+            raise AssertionError('The command is expected to fail but it doesn\'.')
+        elif not expect_failure and self.exit_code != 0:
+            raise AssertionError('The command failed. Exit code: {}'.format(self.exit_code))
+
+        self.json_value = None
+
+    def get_output_in_json(self):
+        if not self.json_value:
+            self.json_value = json.loads(self.output)
+
+        if not self.json_value:
+            raise AssertionError('The command output cannot be parsed in json.')
+
+        return self.json_value
+
+    def _in_process_execute(self, command):
+        # from azure.cli import  as cli_main
+        from azure.cli.main import main as cli_main
+        from six import StringIO
+        from vcr.errors import CannotOverwriteExistingCassetteException
+
+        if command.startswith('az '):
+            command = command[3:]
+
+        output_buffer = StringIO()
+        try:
+            # issue: stderr cannot be redirect in this form, as a result some failure information
+            # is lost when command fails.
+            self.exit_code = cli_main(shlex.split(command), file=output_buffer) or 0
+            self.output = output_buffer.getvalue()
+        except CannotOverwriteExistingCassetteException as ex:
+            raise AssertionError(ex)
+        except CliExecutionError as ex:
+            raise AssertionError(ex)
+        except Exception as ex:  # pylint: disable=broad-except
+            self.exit_code = 1
+            self.output = output_buffer.getvalue()
+            self.process_error = ex
+        finally:
+            output_buffer.close()
+
+    def _out_of_process_execute(self, command):
```

Comment:
Not used for now. I was experimenting run command in separate shell so as to mimic a real scripting environment.